app/jobs/notify_matches_by_sms_job.rb

```
@@ -3,8 +3,10 @@ class NotifyMatchesBySmsJob < ApplicationJob
 
   LEAD_TIME = 90.minutes # we start sending sms X minutes before campaign ends
 
-  def perform(campaign)
-    Rails.logger.info("Run NotifyMatchesBySmsJob for campaign_id #{campaign.id}")
+  def perform(campaign_id)
+    Rails.logger.info("Run NotifyMatchesBySmsJob for campaign_id #{campaign_id}")
+    campaign = Campaign.find_by(id: campaign_id)
```

Pour info avec Rails, si tu veux find by id, tu peux écrire juste `Campaign.find(campaign_id)` 🙂

---

ola @colinemarie :sun_with_face: 

Le `find_by(id:` (ou `find_by_id`) permet de renvoyer `nil` si la campagne a été détruite entre temps, et pas une `ActiveRecord::NotFound`.

Peut-être qu'on veut que ça raise une erreur, à voir !

---

Aaaah merci, je me disais que tu avais une bonne raison 😀

---

Je pense que c’est mieux d utiliser `find` pour fail early et d’avoir une erreur explicite car de toute façon si campaign est nil ça va planter ensuite

---

:white_check_mark: